### ISSUE TITLE
Fetching Jenkins CLI jar

### ISSUE THREAD

**Issue by ericgoedtel** [chef-cookbooks/jenkins#512]
### Cookbook version

2.6.0
### Chef-client version

Chef 12.13.37
### Platform Details

Ubuntu 14.04 on AWS EC2
### Scenario:

Configuring a Jenkins user through the jenkins_user resource
### Steps to Reproduce:

Start with the following recipe:

```
include_recipe 'jenkins::master'
jenkins_user 'grumpy' do
  full_name    'Grumpy Dwarf'
  email        'grumpy@example.com'
  public_keys  ['ssh-rsa ...']
end
```

Execute the recipe
### Expected Result:

A user is created in Jenkins
### Actual Result:

An HTTP 503 occurs when attempting to download the Jenkins CLI jar:

```
Net::HTTPFatalError
-------------------
remote_file[/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/jenkins-cli.jar] (dynamically defined) had an error: Net::HTTPFatalError: 503 "Service Unavailable"

Cookbook Trace:
---------------
/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/jenkins/libraries/_helper.rb:436:in `ensure_cli_present!'
/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/jenkins/libraries/_helper.rb:62:in `executor'
/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/jenkins/libraries/user.rb:142:in `current_user'
/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/jenkins/libraries/user.rb:75:in `load_current_resource'
/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
/var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'

Resource Declaration:
---------------------
# In /var/chef/runs/72ca5285-ea20-48da-9628-a95b97fb2800/local-mode-cache/cache/cookbooks/jenkins-layer/recipes/setup.rb

67 jenkins_user 'grumpy' do
68   full_name    'Grumpy Dwarf'
69   email        'grumpy@example.com'
70   public_keys  ['ssh-rsa ...']
71 end
```

Subsequent runs result in this working--so the problem seems to be that Jenkins is starting too slowly for the Chef recipe. It seems like the resource is hitting `remote_file` and it does not have any retries configured.

**Comment by bclow-invoca** [comment]
The `503 "Service Unavailable"` error shows up when running a vagrant as well. Makes testing next to impossible.

**Comment by iennae** [comment]
I believe this is related to using Jenkins 2.0 or later. We are still working on making this cookbook Jenkins 2.0 compatible. If you try setting the attribute  to something that is pre 2.0, for example master['version'] = '1.658' do you still see the issue?

**Comment by severi** [comment]
I'm experiencing the same problem with (at least) master['version'] = '1.658' when using cookbook version 4.1.2

**Comment by iennae** [comment]
Interesting, I wasn't running into this issue so good to have additional information about impact. Thank you!

**Comment by severi** [comment]
Update regarding this. It seems that this issue does not occur if I set the following:

node.normal['jenkins']['master']['jvm_options'] = '-Djenkins.install.runSetupWizard=false'

**Comment by iennae** [comment]
Weird as that is one of the 2.0 issues, basically making sure the setup wizard doesn't get set up.

**Comment by Dispader** [comment]
```
node.normal['jenkins']['master']['jvm_options'] = '-Djenkins.install.runSetupWizard=false'
```

gets me past the '503' error, as well.  Thank you so much, sir!

**Comment by cheeseplus** [comment]
Closing this as we've merged #557
